underscore.js

```diff
@@ -533,12 +533,20 @@
   // Produce a duplicate-free version of the array. If the array has already
   // been sorted, you have the option of using a faster algorithm.
   // Aliased as `unique`.
-  _.uniq = _.unique = function(array, isSorted, iteratee, context) {
+  _.uniq = _.unique = function(array, isSorted, iteratee, context, predicate) {
     if (!_.isBoolean(isSorted)) {
       context = iteratee;
       iteratee = isSorted;
       isSorted = false;
+      predicate = _.isNull(predicate) ? context : predicate;
```

Comment:
yes, that would be better to ensure that context is a function before assigning it to predicate. thanks.